crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -0,0 +1,91 @@
+//! Protocol fee implementation.
+//!
+//! The protocol fee is a fee that is defined by the protocol and for each order
+//! we define the way to calculate the protocol fee based on the configuration
+//! parameters.
+
+use crate::{
+    boundary::{self},
+    domain,
+};
+
+/// Constructs fee policies based on the current configuration.
+#[derive(Debug)]
+pub struct ProtocolFee {
+    policy: Policy,
+    fee_policy_skip_market_orders: bool,
+}
+
+impl ProtocolFee {
+    pub fn new(policy: Policy, fee_policy_skip_market_orders: bool) -> Self {
+        Self {
+            policy,
+            fee_policy_skip_market_orders,
+        }
+    }
+
+    /// Get policies for order.
+    pub fn get(&self, order: &boundary::Order, quote: Option<&domain::Quote>) -> Vec<Policy> {
+        match order.metadata.class {
+            boundary::OrderClass::Market => {
+                if self.fee_policy_skip_market_orders {
+                    vec![]
+                } else {
+                    vec![self.policy]
+                }
+            }
+            boundary::OrderClass::Liquidity => vec![],
+            boundary::OrderClass::Limit => {
+                if !self.fee_policy_skip_market_orders {
+                    return vec![self.policy];
+                }
+
+                // if the quote is missing, we can't determine if the order is outside the
+                // market price so we protect the user and not charge a fee
+                if quote.is_none() {
```

Comment:
nit: This gets  rid of the unwraps further down.
```suggestion
                let Some(quote) = quote else {
```

Reply:
Nice